src/server/session.ts

```diff
@@ -93,6 +93,9 @@ namespace ts.server {
         export const Completions = "completions";
         export const CompletionsFull = "completions-full";
         export const CompletionDetails = "completionEntryDetails";
+        export const IsCompileOnSaveEnabledForProject = "isCompileOnSaveEnabledForProject";

```

Comment:
I'm curious if you need this message? It can be collapsed with `CompileOnSaveAffectedFileList` so if compile on save is disabled - response to `CompileOnSaveAffectedFileList` will indicate it